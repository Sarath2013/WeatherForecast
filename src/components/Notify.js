import React from 'react';
import PropTypes from 'prop-types';

const Notify = (props) => {
    if (props.error)
        console.error(props.error);
    return (
        <div className="center-content notify-content">
            <h1 className={props.classVal}>{props.message}</h1>
        </div>
    );
}

Notify.propTypes = {
    message: PropTypes.string.isRequired,
    classVal: PropTypes.string,
    error: PropTypes.object
}

export default Notify;