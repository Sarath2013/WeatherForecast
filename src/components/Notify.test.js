import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Notify from './Notify';

let wrapper;

describe("Unit Testing - Rendering Notify with No weather forecasts", () => {
    beforeAll(() => {
    });

    afterAll(() => {
        wrapper = null
    });

    test('renders snapshot', () => {
        wrapper = shallow(<Notify message="No weather forecasts avaialble." />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('ensure notify detials', () => {
        expect(wrapper.find('h1').text()).toEqual('No weather forecasts avaialble.');
    });
});

describe("Unit Testing - Rendering Notify with Error", () => {
    beforeAll(() => {
    });

    afterAll(() => {
        wrapper = null
    });

    test('renders snapshot', () => {
        wrapper = shallow(<Notify classVal="alert-red" message="Something went wrong. Sorry, we can't show weather information at this time." />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('ensure notify details', () => {
        expect(wrapper.find('h1').text()).toEqual("Something went wrong. Sorry, we can't show weather information at this time.");
        expect(wrapper.find('h1').props().className).toEqual("alert-red");
    });
});