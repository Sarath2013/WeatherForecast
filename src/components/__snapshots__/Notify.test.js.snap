// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Testing - Rendering Notify with Error renders snapshot 1`] = `
<div
  className="center-content notify-content"
>
  <h1
    className="alert-red"
  >
    Something went wrong. Sorry, we can't show weather information at this time.
  </h1>
</div>
`;

exports[`Unit Testing - Rendering Notify with No weather forecasts renders snapshot 1`] = `
<div
  className="center-content notify-content"
>
  <h1>
    No weather forecasts avaialble.
  </h1>
</div>
`;
